### PR TITLE
Fix displaying recordings over an hour

### LINKED ISSUE #4143: Audio recordings longer than an hour display incorrect length
#### Software and hardware versions 
Collect v1.28.1 (and earlier)

#### Problem description

Audio recordings longer than an hour only display the number of minutes past the hour. So, if you record for 1 hour and 14 seconds the length of the recording would be "00:14":

69143b791eef27e2a3d60a7db8fbcfbe2c74acda

#### Steps to reproduce the problem

1. Open a form with an audio question
2. Choose an audio file longer than an hour (easier than recording)

#### Expected behavior

If the time is greater than an hour we should probably display the time as "hh:mm:ss:". We should make sure this works for 100+ hours (although I doubt we'll see anyone record for that long).

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4143

#### What has been done to verify that this works as intended?

Added tests.

#### Why is this the best possible solution? Were any other approaches considered?

I was surprised I couldn't find a solution for this in JodaTime! Not a big deal to have a custom implementation of this as it doesn't need a lot of test coverage.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Only part of the app that's changed is the audio widget and how it displays the position and duration.

#### Do we need any specific form for testing your changes? If so, please attach one.

Any form with an audio question will do!

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/audio/AudioControllerView.java [hunk 1]
```diff
@@ -101,10 +103,6 @@ private void onPositionChanged(Integer newPosition) {
         }
     }
 
-    private static String getTime(long seconds) {
-        return new DateTime(seconds, DateTimeZone.UTC).toString("mm:ss");
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/audio/AudioControllerView.java:105]: What happens if you just do HH:mm:ss here? Does it fail beyond 24 hours? Does that matter?

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/audio/AudioControllerView.java:105]: Yeah beyond 24 hours we roll back over. It's probably not a huge problem as I don't imagine people are recording/choosing audio that long but using `DateTime` is a little weird.

Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/audio/AudioControllerView.java:105]: Oh, I see. It also always puts the hours which is not what we want for clips under an hour.

**Review by lognaturel — APPROVED** [review]

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android 7.0, 8.1, 10.0

Verified cases:
* added ~1h audio
* added ~2h audio
* added audio from GD
* added audio from device
* recorded ~1h audio
* removed response from a new option in player
* removed response from dialog
* light / dark / magenta mode
 
Additionally, I took a look at the new audio player. Audio player remembers where audio left off before rotation -> after the second rotation, it looks like audio goes back to the beginning but when a user plays it, it is continued -> next rotations cause that audio goes back to the beginning. This behavior seems to be better than before where audio goes back to the beginning after each rotation

**Comment by lognaturel** [comment]
Thanks, @mmarciniak90! Since we're going to be doing a lot more QA on audio recording soon, I don't think we need to verify other Android versions for this one. I think we're ready to merge.

**Comment by mmarciniak90** [comment]
> I don't think we need to verify other Android versions for this one.

OK!

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**